app/components/blacklight/search_context/server_item_pagination_component.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module Blacklight
+  module SearchContext
+    class ServerItemPaginationComponent < Blacklight::Component
+      with_collection_parameter :search_context
+
+      def initialize(search_context:, search_session:, current_document:)
+        @search_context = search_context
+        @search_session = search_session
+        @current_document_id = current_document.id
+      end
+
+      def render?
+        @search_context.present? && (@search_context[:prev] || @search_context[:next])
```

Comment:
Blacklight 8 has this as:

```suggestion
        @search_context.present? && (@search_context[:prev] || @search_context[:next] || total.positive?) && (@search_session['document_id'] == @current_document_id)
```

Should this match, or is there a reason this is different?

Reply:
I tried that version and it wasn't compatible with Blacklight 7.

My hope with this PR is that we will be able to add a component to replace our custom pagination partial that will work with both Blacklight 7 and Blacklight 8 without significant revisions when upgrading between versions.